### PR TITLE
create friends list panel and add Zezima to it

### PR DESCRIPTION
**Pull request by EthanVieira**
image

### CODE REVIEW

File: src/assets/ui/buttons/DeleteFriendButton.png
Comment (CurtisGreen) [inline src/assets/ui/buttons/DeleteFriendButton.png:1]: Looks kinda fuzzy in compared to Add Friend

File: src/assets/ui/buttons/FriendsButton.png
Comment (CurtisGreen) [inline src/assets/ui/buttons/FriendsButton.png:1]: :)

File: src/cookie-io.js [hunk 1]
```diff
@@ -114,6 +114,21 @@ class CharacterData {
...
 
+    addFriend(name) {
+        if (!this.characterData.friends.includes(name)) {
+            this.characterData.friends.push(name);
+        }
+    }
+    removeFriend(name) {
+        const index = this.characterData.friends.indexOf(name);
+        if (index > -1) {
+            this.characterData.friends.splice(index, 1);
```
Comment (CurtisGreen) [inline src/cookie-io.js]: TIL `splice`

Could also do `this.characterData.friends = this.characterData.friends.filter((friend) => friend != name);`
No preference either way

File: src/ui/panels/friends.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        this.friends.forEach((name) => {
+            // TODO: when we add functionality for on/offline
+            // we would change text color here
+
+            let textObj = this.scrollWindow.add
+                .text(0, 0, "", { font: "15.5px runescape" })
+                .setInteractive()
+                .on("pointerup", () => {
+                    // TODO: maybe allow message sending if they are online.
```
Comment (CurtisGreen) [inline src/ui/panels/friends.js:87]: 😂

Comment (CurtisGreen) [reply src/ui/panels/friends.js:87]: Ah, one line off. Was laughing at chatGPT zezima

File: src/cookie-io.js (hunk 1 above)
Comment (EthanVieira) [reply src/cookie-io.js]: Updated to use this!

File: src/assets/ui/buttons/DeleteFriendButton.png
Comment (EthanVieira) [reply src/assets/ui/buttons/DeleteFriendButton.png:1]: I think I fixed this

**Review by CurtisGreen — APPROVED** [review]

**Comment by EthanVieira** [comment]
[Zezima on friends list](https://trello.com/c/BHHtn6ew/11-zezima-on-friends-list)

**Pull request merged**